lib/ui/surveydetail/survey_detail_view_model.dart

```diff
@@ -40,6 +49,97 @@ class SurveyDetailViewModel extends StateNotifier<SurveyDetailState> {
     state = SurveyDetailState.error(result.getErrorMessage());
   }
 
+  void saveDropdownAnswer(String questionId, Answer answer) {
+    _saveAnswersToQuestions(questionId, [SubmitAnswer.fromAnswer(answer)]);
+  }
+
+  List<Answer>? _getAnswers(String questionId) {
+    return _surveySubject.value.questions
+        .firstWhereOrNull((element) => element.id == questionId)
+        ?.answers;
+  }
+
+  void saveRating(String questionId, int rating) {
+    final answers = _getAnswers(questionId);
+    final selectedAnswer = answers
+        ?.firstWhereOrNull((element) => element.displayOrder == rating - 1);
+
+    final submitAnswers = selectedAnswer != null
+        ? [SubmitAnswer.fromAnswer(selectedAnswer)]
+        : <SubmitAnswer>[];
+
+    _saveAnswersToQuestions(questionId, submitAnswers);
+  }
+
+  void saveMultiChoiceItems(
+    String questionId,
+    List<MultiChoiceItem> items,
+  ) {
+    _saveAnswersToQuestions(questionId,
+        items.map((e) => SubmitAnswer(id: e.id, answer: e.label)).toList());
+  }
+
+  void saveText(String questionId, String text) {
+    final answers = _getAnswers(questionId);
+    if (answers == null || answers.isEmpty) return;
+
+    final submitAnswers = text.isNotEmpty
+        ? [SubmitAnswer(id: answers.first.id, answer: text)]
+        : <SubmitAnswer>[];
+    _saveAnswersToQuestions(questionId, submitAnswers);
+  }
+
+  void saveTextFieldsAnswer(String questionId, String answerId, String text) {
+    final submitQuestion = _submitQuestions
+        .firstWhereOrNull((element) => element.id == questionId);
+    final submitAnswer = submitQuestion?.answers
+        .firstWhereOrNull((element) => element.id == answerId);
+
+    if (text.isNotEmpty) {
+      final newSubmitAnswer = SubmitAnswer(id: answerId, answer: text);
+      if (submitQuestion == null) {
+        _submitQuestions
+            .add(SubmitQuestion(id: questionId, answers: [newSubmitAnswer]));
+      } else if (submitAnswer == null) {
+        submitQuestion.answers.add(newSubmitAnswer);
+      } else {
+        submitAnswer.answer = text;
+      }
+    } else if (submitQuestion != null && submitAnswer != null) {
+      submitQuestion.answers.removeWhere((element) => element.id == answerId);
+    }
```

Comment:
What do you think about improving this code a bit for more readable? 🙏 
```suggestion
    final submitQuestion = _submitQuestions
        .firstWhereOrNull((element) => element.id == questionId);
    final submitAnswer = submitQuestion?.answers
        .firstWhereOrNull((element) => element.id == answerId);

   // Clear answers
    if (text.isEmpty && submitQuestion != null && submitAnswer != null) {
      submitQuestion.answers.removeWhere((element) => element.id == answerId);
      return;
    }
    
    final newSubmitAnswer = SubmitAnswer(id: answerId, answer: text);
    if (submitQuestion == null) {
      _submitQuestions
          .add(SubmitQuestion(id: questionId, answers: [newSubmitAnswer]));
        return;
    }
    
    if (submitAnswer == null) {
      submitQuestion.answers.add(newSubmitAnswer);
      return;
    } 
    
    submitAnswer.answer = text;
```

Reply:
Resolved in https://github.com/nimblehq/ic-flutter-avishek/pull/69/commits/7bf1b8cd4831b1304336e0d42a261965537e9515